packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -81,19 +104,239 @@ typedef ActionListenerCallback = void Function(Action<Intent> action);
+  /// if one exists.
+  ///
+  /// When invoked, the resulting [Action] tries to find the closest [Action] in
+  /// the given `context` that handles the same type of [Intent] as the
+  /// `defaultAction`, then calls its [Action.invoke] method. When no override
+  /// [Action]s can be found, it invokes the `defaultAction`.
+  ///
+  /// An overridable action delegates everything to its override if one exists,
+  /// and has the same behavior as its `defaultAction` otherwise. For this
+  /// reason, the override has full control over whether and how an [Intent]
+  /// should be handled, or a key event should be consumed. An override
+  /// [Action]'s [callingAction] property will be set to the [Action] it
+  /// currently overrides, giving it access to the default behavior. See the
+  /// [callingAction] property for an example.
+  ///
+  /// The `context` argument is the [BuildContext] to find the override with. It
+  /// is typically a [BuildContext] above the [Actions] widget that contains
+  /// this overridable [Action].
+  ///
+  /// The `defaultAction` argument is the [Action] to be invoked where there's
+  /// no ancestor [Action]s can't be found in `context` that handle the same
+  /// type of [Intent].
+  ///
+  /// This is useful for providing a set of default [Action]s in a leaf widget
+  /// to allow further overriding, or to allow the [Intent] to propagate to
+  /// parent widgets that also support this [Intent].
+  ///
+  /// {@tool snippet --template=freeform}
+  /// This sample implements a custom text input field that handles the
+  /// [DeleteTextIntent] intent, as well as a US telephone number input widget
+  /// that consists of multiple text fields for area code, prefix and line
+  /// number. When the backspace key is pressed, the phone number input widget
+  /// sends the focus to the preceding text field when the currently focused
+  /// field becomes empty.
+  ///
+  /// ```dart preamble
+  /// // This implements a custom phone number input field that handles the
+  /// // [DeleteTextIntent] intent.
+  /// class DigitInput extends StatefulWidget {
+  ///   const DigitInput({
+  ///     Key? key,
+  ///     required this.controller,
+  ///     required this.focusNode,
+  ///     this.maxLength,
+  ///     this.textInputAction = TextInputAction.next,
+  ///   }) : super(key: key);
+  ///
+  ///   final int? maxLength;
+  ///   final TextEditingController controller;
+  ///   final TextInputAction textInputAction;
+  ///   final FocusNode focusNode;
+  ///   @override
+  ///   DigitInputState createState() => DigitInputState();
+  /// }
+  ///
+  /// class DigitInputState extends State<DigitInput> {
+  ///   late final Action<DeleteTextIntent> _deleteTextAction = CallbackAction<DeleteTextIntent>(
+  ///     onInvoke: (DeleteTextIntent intent) {
+  ///       // For simplicity we delete everything in the section.
+  ///       widget.controller.clear();
```

Comment:
Unfortunately it's not that easy to implement an `Action` for `DeleteTextIntent` or `SelectAllTextIntent` right now so I had to use  this.